com.spotify/index.js

```diff
@@ -0,0 +1,662 @@
+      console.log("searching for  " + searchURL);
+      return Tp.Helpers.Http.get(searchURL, {
+          accept: 'application/json',
+          extraHeaders: {
+              'Authorization': 'Bearer ' + this.state.accessToken
+          }
+      }).then(response => {
+          const parsed = JSON.parse(response);
+          console.log(parsed.tracks);
+          return parsed;
+      });
+  }
+
+
+  do_play_song ({toPlay}){   //TODO : DELETE THIS LATER
+    this.search(toPlay,'track').then(searchResults=>{
+        //TODO: Possibly add more error checking here.
+        if (!searchResults.hasOwnProperty('tracks')) return;
+        let data = {'uris': [searchResults.tracks.items[0].uri]};
+        console.log("data is " + JSON.stringify(data));
+        return this.player_play_helper(JSON.stringify(data));
+    });
+  }
+
+  do_play_something({toPlay}){   // TODO : DELETE THIS LATER
+    this.do_play_song({toPlay});
+  }
+
+  do_play_album({toPlay}){
+    this.search(toPlay,'album').then(searchResults=>{
+      //TODO: Possibly add more error checking here.
+      if (!searchResults.hasOwnProperty('albums')) return;
+      if(searchResults.albums.total == 0) return;
+      let data = {'context_uri':searchResults.albums.items[0].uri};
+      console.log('data is ' + JSON.stringify(data));
+      return this.player_play_helper(JSON.stringify(data));
+    });
+  }
+
+  do_play_artist({toPlay}){
+    this.search(toPlay,'artist').then(searchResults=>{
+      //TODO: Possibly add more error checking here.
+      if (!searchResults.hasOwnProperty('artists')) return;
+      if(searchResults.artists.total == 0) return;
+      let data = {'context_uri':searchResults.artists.items[0].uri};
+      console.log('data is ' + JSON.stringify(data));
+      return this.player_play_helper(JSON.stringify(data));
+    });
+  }
+
+  do_play_playlist({toPlay}){
+    this.search(toPlay,'playlist').then(searchResults=>{
+      //TODO: Possibly add more error checking here.
+      if (!searchResults.hasOwnProperty('playlists')) return;
+      if (searchResults.playlists.total == 0) return;
+      let data = {'context_uri':searchResults.playlists.items[0].uri};
+      console.log('data is ' + JSON.stringify(data));
+      return this.player_play_helper(JSON.stringify(data));
+    });
+  }
+
+  splitMultiString(string){
+    console.log('song string is ' + string);
+    string = string.toString().toLowerCase();
+    // string can be split by the word comma or actual ','
+    string = string.replace(new RegExp(' comma ','g'),',');
+    let songs = string.split(',');
+    return songs;
+  }
+
+  async songNamesToURIs(songs){
+    let uris = [];
+    for (let i = 0; i < songs.length; i++){
+      let searchResults = await this.search(songs[i].trim(),'track');
+      //TODO: Possibly add more error checking here.
+      if (!searchResults.hasOwnProperty('tracks') || searchResults.tracks.total === 0) continue;
+      uris.push(searchResults.tracks.items[0].uri);
+    }
```

Comment:
If you write this as
```
songNamesToURIs(songs) {
return Promise.all(songs.map((s) => this.search(s, 'track')));
}
```
the searches will be executed in parallel, which is likely to be faster.